### DISCUSSION TITLE
Let's discuss about OSM and martin

### DISCUSSION THREAD

**Discussion by sharkAndshark** [maplibre/martin#980] [Ideas]
I'm not sure, one main concern of users is how to use `martin` and `OSM` data together?   
## Better document and learning curve
* Maybe add a section in `martin book` to demonstrate the combination of  `OSM` data and `martin` , and even other `FOOS4G` things like,  like `osm2pgsql` and  `open tile schema`?
* The doc should be easy to follow, step by step.
* Better if some out of box things included in doc, like `docker compose` template

A section about how to serve a OSM base map of one city? I'm not sure.  

## Further more
Maybe martin could do some work to simplify its combination of OSM data in future, only if possible and reasonable.

**Comment by nyurik** [comment]
Fundamentally, this is not a question of Martin, but of how to convert OSM data into a database.  I would recommend using osm2pgsql - download some OSM extract, get just the data needed into a postgres DB  (e.g. using a docker container), and then point martin to the generated database.

**Reply by sharkAndshark** [reply]
Agree. Maybe it's better to have some docs or post about this.

**Reply by jjcfrancisco** [reply]
Big topic but another option is [Imposm3](https://imposm.org/docs/imposm3/latest/). It gets your `osm.pbf` files and places them into a database - OpenMapTiles uses it under the hood.

**Reply by nyurik** [reply]
@jjcfrancisco to my knowledge, based on my conversation with Imposm's author, this project is no longer actively developed, nor do they want to accept any significant improvements.  I would rather not promote projects like that, even though i do think imposm has many great features.

**Reply by jjcfrancisco** [reply]
Fair. I was not aware of such thing. I suppose `osm2pgsql` is the only other option - there's certainly a great need for a CLI type tool similar to OpenMaptiles/Imposm.

**Reply by nyurik** [reply]
yeah, i think with the recent dev investment made to osm2pgsql, it became fairly good to cover most required use cases

**Reply by jjcfrancisco** [reply]
Re new investment for `osm2pgsql`: are there plans to do `yaml` mappings like in `Imposm3`? It is a useful option - hence the popularity of OpenMapTiles schema. `tilemaker` follows a similar thing but with `mbtiles` as only output. This will be important even more so if `Imposm3` is not looking to further develop things... OpenMapTiles relies **heavily** on it.
